src/vs/workbench/api/common/extHost.protocol.ts

```diff
@@ -113,6 +113,8 @@ export interface MainThreadClipboardShape extends IDisposable {
 
 export interface MainThreadCommandsShape extends IDisposable {
 	$registerCommand(id: string): void;
+	$onDidExecuteCommand(): void;
+	$disposeListener(): void;
```

Comment:
For clarity I would call these two `registerCommandListener` and `unregisterCommandListener`